dashboard/src/controllers/widgets/user-role-widget-controller.js

```diff
@@ -98,42 +94,77 @@ window.Discotron.UserRoleWidgetController = class extends window.Discotron.Widge
             button.disabled = true;
             let value = input.value;
             input.value = "";
-            if (this._guild === undefined) {
-                // The text is a user ID, we try to send it directly
-                Discotron.User.get(value).then((user) => {
-                    this._addUserEntry(user);
-                }).catch(console.error);
-            } else {
-                // The text is the name of a role or a user
-                this._addEntry(value);
-            }
+
+            // The text is the name of a role or a user
+            this._addEntry(value);
         };
     }
 
     /**
-     * Activates the add button if the name corresponds to 
+     * Activates the add button if the name corresponds to a valid discord name, or a valid user id
+     * @param {string} name Username of a discord user or role name
+     */
+    _checkIdValidity(id) {
+        let button = document.querySelector("#add-button");
+        button.disabled = true;
+
+        // This was not an existing user, check if it's an id
+        if (id.match(/^[0-9]*$/)) {

```

Comment:
Replace the * with a + (else an empty string would *theoretically* be accepted here).